tests/src/rules/no-extraneous-dependencies.js

```diff
@@ -312,3 +313,35 @@ ruleTester.run('no-extraneous-dependencies', rule, {
     }),
   ],
 })
+
+describe('TypeScript', function () {
+  getTSParsers()
+    .filter((parser) => parser !== require.resolve('typescript-eslint-parser'))
```

Comment:
why filter this one out?

Reply:
`typescript-eslint-parser` is the old parser and doesn't support the `importKind` property. That parser was not done at the time of typescript 3.8 which is the one that supports this feature

I've added it back with the cases as all invalid.